File: android/ui/src/main/java/com/trafi/ui/theme/Spacing.kt
@file:Suppress("unused")

package com.trafi.ui.theme

import androidx.compose.runtime.Immutable
import androidx.compose.runtime.Stable
import androidx.compose.ui.unit.Dp
import com.trafi.ui.theme.internal.SpacingScale

object Spacing {
    @Stable val xxxs: Dp = SpacingScale.xxxs
    @Stable val xxs: Dp = SpacingScale.xxs
    @Stable val xs: Dp = SpacingScale.xs
    @Stable val sm: Dp = SpacingScale.sm
    @Stable val md: Dp = SpacingScale.md
    @Stable val lg: Dp = SpacingScale.lg
    @Stable val xl: Dp = SpacingScale.xl
    @Stable val xxl: Dp = SpacingScale.xxl
    @Stable val xxxl: Dp = SpacingScale.xxxl
}

@Immutable
data class MaasSpacing(
    val globalMargin: Dp = SpacingScale.Default.GlobalMargin
)

File: ios/Sources/MaaSTheme/Spacing.swift
import Foundation
import MaasCore

public struct Spacing {

    public let value: Double

    public static var xxxs: Spacing { .init(value: Double(SpacingScale().xxxs)) }
    public static var xxs: Spacing { .init(value: Double(SpacingScale().xxs)) }
    public static var xs: Spacing { .init(value: Double(SpacingScale().xs)) }
    public static var sm: Spacing { .init(value: Double(SpacingScale().sm)) }
    public static var md: Spacing { .init(value: Double(SpacingScale().md)) }
    public static var lg: Spacing { .init(value: Double(SpacingScale().lg)) }
    public static var xl: Spacing { .init(value: Double(SpacingScale().xl)) }
    public static var xxl: Spacing { .init(value: Double(SpacingScale().xxl)) }
    public static var xxxl: Spacing { .init(value: Double(SpacingScale().xxxl)) }

    public static var `default`: Spacing {
        .init(value: Double(SpacingScale.Default().GlobalMargin))
    }
}
